File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Settings/SettingSwitchCell.m
//
//  SettingSwitchCell.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "SettingSwitchCell.h"
#import "SwitchItemModel.h"

@interface SettingSwitchCell()

@property (weak, nonatomic) IBOutlet UISwitch *switchCtrl;

@end

@implementation SettingSwitchCell

- (void)setModel:(SwitchItemModel *)model {
    
    [super setModel:model];
    self.switchCtrl.on = model.on;
}

- (IBAction)valueChanged:(UISwitch *)sender {
    
    SwitchItemModel *model = (id)self.model;
    model.on = sender.on;
    [self.delegate cell:self didChageModel:self.model];
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Settings/SettingSwitchCell.swift
//
//  SettingSwitchCell.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 12/11/18.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

class SettingSwitchCell: BaseSettingsCell {
    //MARK: - IBOutlets
    @IBOutlet private weak var switchCtrl: UISwitch!
    //MARK: - Properties
    var modelSwitch = SwitchItemModel()
    
    //MARK: - Overrides
    override func updateModel(_ model: BaseItemModel) {
        guard let model = model as? SwitchItemModel else { return }
        modelSwitch = model
        switchCtrl.isOn = modelSwitch.on
        label.text = modelSwitch.title
    }
    
    //MARK: - Actions
    @IBAction func valueChanged(_ sender: UISwitch) {
        modelSwitch.on = sender.isOn
        delegate?.cell(self, didChageModel: modelSwitch)
    }
}
